File: test-suite/src/test/java/io/micronaut/email/docs/TemplateBodyDecoratorTest.java
package io.micronaut.email.docs;

import io.micronaut.context.BeanContext;
import io.micronaut.email.template.TemplateBodyDecorator;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
public class TemplateBodyDecoratorTest {

    @Inject
    BeanContext beanContext;

    @Test
    void beanOfTypeTemplateBodyDecoratorExists() {
        assertTrue(beanContext.containsBean(TemplateBodyDecorator.class));
    }
}

File: test-suite-python/src/test/python/micronaut/email/docs/TemplateBodyDecoratorTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context import BeanContext
from micronaut.email.template import TemplateBodyDecorator
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test


@MicronautTest(startApplication=False)
class TemplateBodyDecoratorTest:
    bean_context: Annotated[BeanContext, Inject]

    @Test
    def bean_of_type_template_body_decorator_exists(self):
        assert self.bean_context.containsBean(TemplateBodyDecorator)
